internal/scti/handlers.go

```diff
@@ -63,53 +62,38 @@ var (
 	once             sync.Once
-	knownLogs        *prometheus.GaugeVec     // origin => value (always 1.0)
-	lastSCTIndex     *prometheus.GaugeVec     // origin => value
-	lastSCTTimestamp *prometheus.GaugeVec     // origin => value
-	reqsCounter      *prometheus.CounterVec   // origin, op => value
-	rspsCounter      *prometheus.CounterVec   // origin, op, code => value
-	rspLatency       *prometheus.HistogramVec // origin, op, code => value
+	knownLogs        metric.Int64Gauge       // origin => value (always 1.0)
+	lastSCTIndex     metric.Int64Gauge       // origin => value
+	lastSCTTimestamp metric.Int64Gauge       // origin => value
+	reqsCounter      metric.Int64Counter     // origin, op => value
+	rspsCounter      metric.Int64Counter     // origin, op, code => value
+	rspLatency       metric.Float64Histogram // origin, op, code => value
 )
 
 // setupMetrics initializes all the exported metrics.
 func setupMetrics() {
 	// TODO(phboneff): add metrics for deduplication and chain storage.
-	knownLogs = promauto.NewGaugeVec(
-		prometheus.GaugeOpts{
-			Name: "known_logs",
-			Help: "Set to 1 for known logs",
-		},
-		[]string{"origin"})
-	lastSCTTimestamp = promauto.NewGaugeVec(
-		prometheus.GaugeOpts{
-			Name: "last_sct_timestamp",
-			Help: "Time of last SCT in ms since epoch",
-		},
-		[]string{"origin"})
-	lastSCTIndex = promauto.NewGaugeVec(
-		prometheus.GaugeOpts{
-			Name: "last_sct_index",
-			Help: "Index of last SCT",
-		},
-		[]string{"origin"})
-	reqsCounter = promauto.NewCounterVec(
-		prometheus.CounterOpts{
-			Name: "http_reqs",
-			Help: "Number of requests",
-		},
-		[]string{"origin", "ep"})
-	rspsCounter = promauto.NewCounterVec(
-		prometheus.CounterOpts{
-			Name: "http_rsps",
-			Help: "Number of responses",
-		},
-		[]string{"origin", "op", "code"})
-	rspLatency = promauto.NewHistogramVec(
-		prometheus.HistogramOpts{
-			Name: "http_latency",
-			Help: "Latency of responses in seconds",
-		},
-		[]string{"origin", "op", "code"})
+	knownLogs = mustCreate(meter.Int64Gauge("tesseract.known_logs",
+		metric.WithDescription("Set to 1 for known logs")))
+
+	lastSCTTimestamp = mustCreate(meter.Int64Gauge("tesseract.last_sct.timestamp",
+		metric.WithDescription("Time of last SCT since epoch"),
+		metric.WithUnit("ms")))
+
+	lastSCTIndex = mustCreate(meter.Int64Gauge("tesseract.last_sct.index",
+		metric.WithDescription("Index of last SCT"),
+		metric.WithUnit("{entry}")))
+
+	reqsCounter = mustCreate(meter.Int64Counter("tesseract.http_request.count",
+		metric.WithDescription("CT HTTP requests")))
+
+	rspsCounter = mustCreate(meter.Int64Counter("tesseract.http_response.count",
+		metric.WithDescription("CT HTTP responses")))
+
+	rspLatency = mustCreate(meter.Float64Histogram("tesseract.http_response.duration",
+		metric.WithDescription("CT HTTP response duration"),
+		metric.WithExplicitBucketBoundaries(otel.LatencyHistogramBuckets...),
+		metric.WithUnit("ms")))
```

Comment:
Let's use seconds here as a unit: it's Opentelemetry's default unit for time. I think we should also rename this metric to `http_request.duration` rather, to align with https://opentelemetry.io/docs/specs/semconv/http/http-metrics/#metric-httpserverrequestduration. I can't tell for sure, but should this be `tesseract.http.request.count` rather?

In all fairness: If I were to write this from scratch, I'd use latency... and if I could not use latency I would use "response.duration". But it looks like people who've thought about this for some time have made other decisions, let's stick to default and embrace Opentelemtry way of doing things.